File: with-notes/2022-06/reverse.rb
=begin
* Date: 6.17.22
* Rank: 8 Kyu (Codewars)
* Link: https://www.codewars.com/kata/57a55c8b72292d057b000594/train/ruby
* In My Own Words: 
    Take a string and reverse the words in that string. Punctuation should remain attached to the original word and the function should return and empty string if given an empty string. A single word string should return the single word. 
* Test Conditions: 
    "Hello World" --> "World Hello"
    "Hi There." --> "There. Hi"
    "Ruby is cool!" --> "cool! is Ruby"
* Pseudocode: 
    1. Handle edge cases (empty string, one word)
    1. Split the string into an array
    2. Reverse the array
    3. Join the array
    4. Return the string
* Notes After Attempt:
=end

def reverse(string)
    return string.split(" ").length == 0 | 1 ? string : string.split(" ").reverse.join(" ")
end

puts (reverse("Ruby is cool!"))

File: with-notes/2022-12/reverse.js
/*
* Date: 12.24.22
* Rank: 7 Kyu
* Link: https://www.codewars.com/kata/5413759479ba273f8100003d/train/javascript
* In My Own Words: 
  Reverse an array without using the built-in method.
* Test Conditions: 
  reverse([1,2,3]), [3,2,1] )
  reverse([1,null,14,"two"]), ["two",14,null,1] )
* Pseudocode: 

* Notes After Attempt:

*/

reverse = function(array) {
  const rev = []
  while (array.length > 0) {
    rev.push(array.splice(array.length-1))
  }
  return rev.flat()
}

module.exports = reverse